app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/summary/SummaryProductFragment.kt

```diff
@@ -461,32 +461,31 @@ class SummaryProductFragment : BaseFragment(), ISummaryProductPresenter.View {
         binding.ecoscoreIcon.setImageResource(product.getEcoscoreResource())
     }
 
-    private fun refreshNutriScorePrompt() {
+    private fun refreshStatesTagsPrompt() {
         //checks the product states_tags to determine which prompt to be shown
         val statesTags = product.statesTags
         showCategoryPrompt = statesTags.contains("en:categories-to-be-completed") && !hasCategoryInsightQuestion
         showNutrientPrompt = statesTags.contains("en:nutrition-facts-to-be-completed") && product.noNutritionData != "on"
-        showEcoscorePrompt = statesTags.contains("en:categories-completed") && product.ecoscore == null
+        showEcoScorePrompt = statesTags.contains("en:categories-completed") && product.ecoscore.isNullOrEmpty().or(product.ecoscore.equals("unknown", true))
```

Comment:
Please use the `||` notation

Reply:
Done. Thanks. Will push the changes once all the conversations are resolved.